src/core/bidi.js

```diff
@@ -158,15 +158,16 @@ function bidi(str, startLevel = -1, vertical = false) {
 
   // Detect the bidi method
   // - If there are no rtl characters then no bidi needed
-  // - If less than 30% chars are rtl then string is primarily ltr
+  // - If less than 30% chars are rtl then string is primarily ltr,
+  //   unless the string is very short.
   // - If more than 30% chars are rtl then string is primarily rtl
   if (numBidi === 0) {
     isLTR = true;
     return createBidiText(str, isLTR);
   }
 
   if (startLevel === -1) {
-    if (numBidi / strLength < 0.3) {
+    if (numBidi / strLength < 0.3 && strLength > 4) {
```

Comment:
I might be missing something obvious here, but wouldn't this mean that _any_ LTR string of length <= 4 would be considered RTL? Let's say we have the string "foo": `numBidi` is 0, so `numBidi / strLength` = 0 / 3 = 0, and `0 < 0.3` holds. However, `3 > 4` doesn't hold, so we end up in the `else` branch and set it to RTL.

Reply:
> Let's say we have the string "foo": `numBidi` is 0, so `numBidi / strLength` = 0 / 3 = 0, and `0 < 0.3` holds. However, `3 > 4` doesn't hold, so we end up in the `else` branch and set it to RTL.

Assuming we actually want this patch, I could also add a new unit-test with an LTR-only string to check exactly the situation described above.

Reply:
@timvandermeij I've added additional unit-tests, please check if those look good; thank you!